### PR TITLE
Minimal changes needed to build an RPM successfully

### PR DESCRIPTION
**Pull request by ndokos**
A couple of changes in agent/Makefile:

- Copy ../lib/configtools if it exists: it will exist when we first
  build the tarball from rpm/Makefile, but it will not exist later
  when we invoke the agent/Makefile again to do the BUILDROOT installation
  from the spec file.

- The spec file needs an install-build-artifacts role to copy the
  generated SHA1 and (optional) SEQNO files.

The unversioned python shebang in ansible/bin/yamlconf caused an RPM build failure. The change
was to reactivate the exclude.pats file and add a couple of patterns to
exclude most of the ansible/ directory from the tarball.

### CODE REVIEW

**Comment by ndokos** [comment]
BTW, the RPM builds with these changes, but I have not actually tested that it works.

File: agent/Makefile [hunk 1]
```diff
@@ -237,7 +237,7 @@ install: install-destdir install-util-scripts install-bench-scripts install-tool
 	${COPY} profile ${DESTDIR}
 	${INSTALL} ${INSTALLOPTS} ${STOCKPILEDIR}
 	${COPY} stockpile ${DESTDIR}
-	${COPY} ../lib/configtools ${LIBDIR}
+	if [ -d ../lib/configtools ] ;then ${COPY} ../lib/configtools ${LIBDIR} ;fi
```
Comment (portante) [inline agent/Makefile:240]: Why is this necessary?  Don't we always have `../lib/configtools`?

File: agent/Makefile [hunk 2]
```diff
@@ -269,5 +269,9 @@ install-tool-scripts: install-destdir
 	cd tool-scripts/postprocess; \
 	${COPY} ${tool-postprocess} ${TOOLDIR}/postprocess
 
+install-build-artifacts:
+	${COPY} SHA1 ${DESTDIR}
+	if [ -f SEQNO ] ;then ${COPY} SEQNO ${DESTDIR} ;fi
```
Comment (portante) [inline agent/Makefile:274]: Aren't these provided in the tar ball at lines 56 and 57 of the `rpm/Makefile`?

File: agent/rpm/exclude.pats [hunk 3]
```diff
@@ -6,3 +6,7 @@ unittests
 tests/
 mock-bin/
 util-scripts/configs/
+ansible/bin
```
Comment (portante) [inline agent/rpm/exclude.pats:9]: Given these patterns, what should be included from the `ansible` subtree in the RPM?

**Review by portante — CHANGES_REQUESTED** [review]
Thanks!

File: agent/Makefile (hunk 1 above)
Comment (ndokos) [reply agent/Makefile:240]: No - the `make install` that the rpm/Makefile does, copies (parts of) the agent/ subdir to some temp directory and then moves the lib/configtool directory from the *top* directory into the agent/lib sudbir in the copy. That's what ends up in the tar ball and that's what the spec file has to deal with. So when the spec file does a `make install` to install the agent/ directory into the BUILDROOT, there is no ../lib/configtools any longer.

File: agent/rpm/exclude.pats (hunk 3 above)
Comment (ndokos) [reply agent/rpm/exclude.pats:9]: AFAICT, the only thing needed is ansible/ara/.

File: agent/Makefile (hunk 2 above)
Comment (ndokos) [reply agent/Makefile:274]: That's what builds them in the local tree, but they have to be copied into the tar ball that is used to build the srpm: otherwise, they are not going to be available to install.

File: agent/Makefile (hunk 1 above)
Comment (portante) [reply agent/Makefile:240]: Removed the need to invoke `make install` in the RPM spec file.

File: agent/Makefile (hunk 2 above)
Comment (portante) [reply agent/Makefile:274]: Removed the need for a special target because we just now copy the entire `agent` tree in place since the "install" used for constructing the source tar ball did the heavy lifting already.

File: agent/rpm/exclude.pats (hunk 3 above)
Comment (portante) [reply agent/rpm/exclude.pats:9]: Removed the need for `exclude.pats` entirely by installing only the pieces we needed from the ansible tree.

**Comment by portante** [comment]
No longer needed.
